splunk/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM docker.io/splunk/splunk:latest
+
+ENV SPLUNK_START_ARGS=--accept-license
+
+EXPOSE 8000
+EXPOSE 8089
+
+USER root
+
+COPY tests/test_logs /opt/splunk/test_logs
+COPY tests/indexes.conf /opt/splunk/etc/apps/search/local/indexes.conf
+COPY scripts/index_verification.sh /opt/splunk/index_verification.sh
+
+RUN microdnf install -y jq && \
+    microdnf clean all
+
+USER ansible
+
+RUN SPLUNK_PASSWORD="Password" /sbin/entrypoint.sh start-service &\
+    sleep 60 &&\
+    bash /opt/splunk/index_verification.sh
```

Comment:
so is that working now without clean shutdown?

Reply:
The index verification itself is working, but running the container fails because AFAIU, the Splunk service is already open. I cannot seem to shut it off

Reply:
```
FAILED - RETRYING: Start Splunk via CLI (5 retries left).
FAILED - RETRYING: Start Splunk via CLI (4 retries left).
FAILED - RETRYING: Start Splunk via CLI (3 retries left).
FAILED - RETRYING: Start Splunk via CLI (2 retries left).
FAILED - RETRYING: Start Splunk via CLI (1 retries left).

TASK [splunk_common : Start Splunk via CLI] ************************************
fatal: [localhost]: FAILED! => {
    "attempts": 5,
    "changed": false,
    "cmd": [
        "/opt/splunk/bin/splunk",
        "start",
        "--accept-license",
        "--answer-yes",
        "--no-prompt"
    ],
    "delta": "0:00:01.044448",
    "end": "2023-06-19 14:25:09.996017",
    "rc": 1,
    "start": "2023-06-19 14:25:08.951569"
}

STDOUT:


Splunk> See your world.  Maybe wish you hadn't.

Checking prerequisites...
	Checking http port [8000]: open
	Checking mgmt port [8089]: open
	Checking appserver port [127.0.0.1:8065]: open
	Checking kvstore port [8191]: open
	Checking configuration... Done.


STDERR:

The CA file specified (/opt/splunk/etc/auth/cacert.pem) does not exist. Cannot continue.
SSL certificate generation failed.


MSG:

non-zero return code
```

Reply:
> The index verification itself is working, but running the container fails because AFAIU, the Splunk service is already open. I cannot seem to shut it off

The service cannot possibly stay open between build time and when you run the container - those are two different containers with different process spaces and network stacks.

As far as the log you quote goes - it seems to me that starting the service causes it to generate certificates for itself that are not getting cleaned up of saved properly because we're not shutting it down in the most orderly way.

What we should probably do is clean the directories back to how they were before we started it before completing the build. One thing you can try is to save a recursive scan of the filenames in `/opt/splunk/etc/` (With E.g. `find`), before and after you run the service and then compare the two file lists.

Also some nitpicks:

- `/opt/splunk/index_verification.sh` is not a good name for what it is doing
- Not sure why you keep living some things out of the script instead of having one organized script to do the whole indexing chore.

Reply:
I went over all the file additions (seems like there are only new files, no changes), over a thousand, removed them all during the build and... still having the same issue. 

as for the nitpicks, - the name is not final and will change, and it is is one script rn, just did not update the PR yet as I'm trying to solve the main issue...

Reply:
Done, it is finally working. 
Building & Running the container now should get us a Splunk container with some logs attached to `test_index` index.